### PR TITLE
docs(*): Add ci script to confirm docs are up to date with commits

### PR DESCRIPTION
**Pull request by cblecker**
Added a script so that CI can verify docs are updated or not. This should improve docs coverage as pre-submit CI will fail if there is a diff in the docs. :)

### CODE REVIEW

**Comment by technosophos** [comment]
This will once again edge us closer to CircleCI's memory and timeout limit. It appears to run okay in the current form... I'll leave it up to @adamreese whether we should add this and risk another round of CircleCI explosions.

**Comment by cblecker** [comment]
If nothing is built, it takes 38s..
```
$ time make verify-docs 
GOBIN=/opt/gopath/src/k8s.io/helm/bin go install  -tags '' -ldflags ' -X k8s.io/helm/pkg/version.Version=v2.3.0 -X k8s.io/helm/pkg/version.GitCommit=dfe2200d01cad3baae4337dd3300213ed682cf95 -X k8s.io/helm/pkg/version.GitTreeState=clean' k8s.io/helm/cmd/...
helm docs up to date.

real	0m38.288s
user	1m37.450s
sys	0m10.845s
```

If the `make build` step is already done, then it adds on about 2 seconds of wall-clock time:
```
$ time make build && time make verify-docs
GOBIN=/opt/gopath/src/k8s.io/helm/bin go install  -tags '' -ldflags ' -X k8s.io/helm/pkg/version.Version=v2.3.0 -X k8s.io/helm/pkg/version.GitCommit=dfe2200d01cad3baae4337dd3300213ed682cf95 -X k8s.io/helm/pkg/version.GitTreeState=clean' k8s.io/helm/cmd/...

real	0m36.676s
user	1m39.475s
sys	0m10.323s
GOBIN=/opt/gopath/src/k8s.io/helm/bin go install  -tags '' -ldflags ' -X k8s.io/helm/pkg/version.Version=v2.3.0 -X k8s.io/helm/pkg/version.GitCommit=dfe2200d01cad3baae4337dd3300213ed682cf95 -X k8s.io/helm/pkg/version.GitTreeState=clean' k8s.io/helm/cmd/...
helm docs up to date.

real	0m2.058s
user	0m0.965s
sys	0m0.798s
```

I assume at the test stage of CircleCI, things are already built, so hopefully this doesn't add too much extra.

File: scripts/ci.sh [hunk 1]
```diff
@@ -27,14 +27,16 @@ run_unit_test() {
...
+    echo "Running 'unit tests'"
     make test-unit
   fi
 }
 
 run_style_check() {
   echo "Running 'make test-style'"
   make test-style
+  echo "Running 'make verify-docs'"
+  make verify-docs
```
Comment (adamreese) [inline scripts/ci.sh]: Run this on a different node

**Review by adamreese — CHANGES_REQUESTED** [review]
Looks great,  thanks for adding this validation!

File: scripts/ci.sh (hunk 1 above)
Comment (cblecker) [inline scripts/ci.sh]: When I tried to run this one a 2) node, it didn't run at all -- I only see a 0 and 1 nodes... that's why I added it here. But maybe I'm not doing it right? I don't have a ton of experience with CircleCI.

Comment (adamreese) [inline scripts/ci.sh]: Looks like I needed to add a third node in circleci.  Give it another shot.

**Review by adamreese — APPROVED** [review]

**Comment by cblecker** [comment]
@adamreese Changes made, and tests pass. :)

**Comment by adamreese** [comment]
Awesome, Thanks!

**Pull request merged**